packages/sockets/src/logic/creators/createSocketCommunicator.ts

```diff
@@ -299,6 +303,36 @@ export default function createSocketCommunicator({
     await receiveArrayPart(buf, 0, recvCount, abort);
   }
 
+  async function reduce(
+    data: unknown[],
+    buf: Array<unknown>,
+    count: number,
+    op: Operator,
+    root: number,
+    abort?: Abort,
+  ) {
+    const isMe = root === selfProcess.rank;
+    const rootProcess = getProcessByRank(root);
+
+    if (!isMe) {
+      // Just send data to root
+      send(data, rootProcess, abort);
```

Comment:
todo(blocking): looks like missing `await`. also early return technic can be applied to avoid nesting